### PR TITLE
spdx-license-list-data: make all formats available

### PR DESCRIPTION
**Pull request by c0bw3b**
###### Motivation for this change

Born out of the discussion in #159182 with @oxzi ;
this change makes the SPDX license list data available in all formats upstream provides but split it into multiple outputs.

Consumer of this data set (such as `platformio`) should pull only the format they need.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by c0bw3b** [comment]
Result of `nixpkgs-review pr 161958` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>esphome</li>
    <li>platformio</li>
    <li>spdx-license-list-data</li>
  </ul>
</details>

---

And in `schema.py` of the new `platformio` derivation:

```session
$ grep "spdx-license" /nix/store/sf4v87wkfzq8m1bdmcfg7pva5c6vvywp-platformio-5.2.4/lib/python3.9/site-packages/platformio/package/manifest/schema.py
        with open("/nix/store/cj3f76bg0q0x5qjiq40zrhvja83271d5-spdx-license-list-data-3.16-json/json/licenses.json") as f:

$ ls -lh /nix/store/cj3f76bg0q0x5qjiq40zrhvja83271d5-spdx-license-list-data-3.16-json/json/
total 252K
dr-xr-xr-x 2 root root  20K  1 janv.  1970 details
dr-xr-xr-x 2 root root 4,0K  1 janv.  1970 exceptions
-r--r--r-- 1 root root  16K  1 janv.  1970 exceptions.json
-r--r--r-- 1 root root 212K  1 janv.  1970 licenses.json
```

File: pkgs/data/misc/spdx-license-list-data/default.nix [hunk 1]
```diff
@@ -2,28 +2,45 @@
...
+  dontBuild = true;
+
   installPhase = ''
     runHook preInstall
 
-    install -vDt $out/json json/licenses.json
+    mkdir -pv $out
+    for t in $_types
+    do
+      _outpath=$(eval echo "\$$t")
```
Comment (mkg20001) [inline pkgs/data/misc/spdx-license-list-data/default.nix:29]:
```suggestion
      _outpath=${!t}
```

Comment (c0bw3b) [reply pkgs/data/misc/spdx-license-list-data/default.nix]: Right! Thanks

**Review by oxzi — APPROVED** [review]
LGTM, thanks for this improvement!

**Comment by oxzi** [comment]
Thanks again. Just fyi, I have rebased #159034 as it kind of depends on this PR.

**Pull request merged**